src/aspire/basis/__init__.py

```diff
@@ -36,6 +36,9 @@ def __init__(self, size, ell_max=None):
         self.count = 0
         self.ell_max = ell_max
         self.ndim = ndim
+        self.dtype = dtype
+        if self.dtype != np.float64:
+            raise NotImplementedError("Currently only implemented for default double (np.float64) type")
```

Comment:
Any reason for using float64 only?

Reply:
Probably we should open a new issue to remember this problem. Including a `dtype` parameter for all related class will be a reasonable way to solve.

Reply:
Yes, an issue is an excellent idea. I will create one. I agree, but for all classes.  If some code is not going to support doubles/singles coherently the other codes will need to warn/error on a mismatch or something. Not sure which is less work tbh...

Reply:
This is now #148

Reply:
For now I think I will punt this to #148